From 0817b0a3243b3c5354827edba5af680334affcdf Mon Sep 17 00:00:00 2001
From: yoav filin <yfilin@habana.ai>
Date: Wed, 26 Oct 2022 13:34:51 +0300
Subject: [PATCH] [SW-108985] drivers: hbm: gaudi: fix hbm row repair flow

fix a sw bug on hbm row repair function.
in the end of row repair a reset to ieee1500 bus
is asserted as it should but not being de-asserted
so the ieee1500 is stuck at reset in case of row repair.
added the de-assertion and print row repair data.

Change-Id: Iccaaecd65d959580c6fd2c18f81ebffc8abf0a1e
Signed-off-by: yoav filin <yfilin@habana.ai>
---

diff --git a/arch/gaudi/arm/hbm.c b/arch/gaudi/arm/hbm.c
index 2e6e07a..831912a 100644
--- a/arch/gaudi/arm/hbm.c
+++ b/arch/gaudi/arm/hbm.c
@@ -1408,6 +1408,7 @@
 #define HBM_ROW_SOFT_REP_DQ_BIT		18
 #define HBM_ROW_SOFT_REP_SID_BIT	19
 #define HBM_ROW_SOFT_REP_EN_BIT		20
+#define HBM_WRSTN_DELAY_US		1
 
 void gaudi_hbm_row_repair(int dev, int dma_pc, int sid, u32 bank, u32 page)
 {
@@ -1458,6 +1459,15 @@
 
 	/* WRSTN assert */
 	RMWREG32(base + GAUDI_HBM_CENTER_CHANNEL * 0x1000 + 0x14, 0x0, 0x1);
+	udelay(HBM_WRSTN_DELAY_US);
+	/* WRSTN de-assert */
+	RMWREG32(base + GAUDI_HBM_CENTER_CHANNEL * 0x1000 + 0x14, 0x1, 0x1);
+
+	preboot_puts_hex_raw("\nHBM ", dev, false);
+	preboot_puts_hex_raw(" performed row repair SUCCESS dma_pc: ", dma_pc, false);
+	preboot_puts_hex_raw(", sid: ", sid, false);
+	preboot_puts_hex_raw(", bank: ", bank, false);
+	preboot_puts_hex(", page: ", page, false);
 }
 
 static int hbm_pc_row_repair(int sid, int device, int ch, int pc,
